integration/tests/cook/test_basic.py

```diff
@@ -43,24 +43,24 @@ def test_basic_submit(self):
             self.assertIsNotNone(job['instances'][0]['sandbox_directory'], message)
 
     def test_no_cook_executor_on_subsequent_instances(self):
-        job_uuid, resp = util.submit_job(self.cook_url, command='exit 1', max_retries=10)
+        job_uuid, resp = util.submit_job(self.cook_url, command='exit 1', max_retries=10) # should launch many instances
         self.assertEqual(resp.status_code, 201, msg=resp.content)
-        try:
-            job = util.wait_for_job(self.cook_url, job_uuid, 'completed', max_wait_ms=60000)
-            message = json.dumps(job, sort_keys=True)
-            self.assertEqual('failed', job['state'], message)
-            self.assertEqual('completed', job['status'], message)
-        except:
-            pass
+        util.wait_until(lambda: util.load_job(self.cook_url, job_uuid, assert_response=False),
+                        lambda job: 'instances' in job and len(job['instances']) > 4, # try and get at least 5 instances
```

Comment:
I'm pretty sure that a job is never without an `instances` field in the response here. Furthermore, even if it is missing, the KeyError is just going to trigger another retry.

If you still want the predicate to be "safe", I think using `get` with a default is simpler than the `in` check:
```python
lambda job: len(job.get('instances', [])) > 4
```